### DISCUSSION TITLE
usocket and ussl https server

### DISCUSSION THREAD

**Discussion by gringtc** [micropython/micropython#14095] [Using MicroPython] [answered]
I'm trying to setup an https server with sockets and ssl in micropython as shown here [http_server_ssl.py](https://github.com/micropython/micropython/blob/2b6f81f2b9d5135774d1b636802194f9666678d5/examples/network/http_server_ssl.py).

When using exactly the example with the given `key` and `cert`, I get this error when connecting to the server via chrome on my laptop:

Logs:
```
Bind address info: [(2, 1, 0, '', ('0.0.0.0', 8443))]
Listening, connect your browser to https://<this_host>:8443/
Client address: ('192.168.0.104', 64191)
Client socket: <socket state=3 timeout=-1 incoming=0 off=0>
```

Error:
```python
Traceback (most recent call last):
  File "<stdin>", line 95, in <module>
  File "<stdin>", line 65, in main
OSError: (-30592, 'MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE')
```

line 65: `client_s = ssl.wrap_socket(client_s, server_side=True, key=key, cert=cert)`

When using my own signed `key` and `cert` like this:
```python
with open('certs/privkey.pem', 'rb') as f:
    key = f.read()
with open('certs/cert.pem', 'rb') as f:
    cert = f.read()
```

I get an invalid key error:
```python
Traceback (most recent call last):
  File "<stdin>", line 69, in <module>
  File "<stdin>", line 39, in main
ValueError: invalid key
```

Any help with setting up the https server would be appreciated. Thanks!

**Comment by Carglglz** [comment, marked as answer]
@gringtc
`OSError: (-30592, 'MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE')`
This is the browser response to a self-signed certificate, to avoid this you need to add the self-signed certificate to the trusted sources in chrome, see https://stackoverflow.com/questions/7580508/getting-chrome-to-accept-self-signed-localhost-certificate

For `ValueError: invalid key` see https://github.com/micropython/micropython/tree/master/tests#test-keycertificates instructions to generate a key/cert pair (MicroPython only supports `DER` format by default, unless using `esp32` port)

**Reply by gringtc** [reply]
Thank You!
